heroku/deploy-a-django-app.md

```diff
@@ -0,0 +1,272 @@
+to read these variables from the environment:
+
+```python
+SECRET_KEY = os.environ['DJANGO_SECRET_KEY']
+DEBUG = False if os.getenv('DJANGO_DEBUG', True) == 'False' else True
+allowed_hosts = os.getenv('DJANGO_ALLOWED_HOSTS', [])
+ALLOWED_HOSTS = allowed_hosts.split(',') if allowed_hosts else []
+```
+
+Make sure to update your app service in `docker-compose.yml` to thread any variables
+that don't have defaults into your local environment:
+
+```yml
+services:
+  app:
+    environment:
+      - DJANGO_SECRET_KEY=really-super-secret
+```
+
+### Configure Django logging
+
+When Gunicorn is running our app on Heroku, we generally want it to log to stdout
+and stderr instead of logging to files, so that we can let Heroku capture the logs
+and see view them with the `heroku logs` CLI command (or in the web console).
+
+By default, Django will not log errors to the console when `DEBUG` is `False`
+([as documented here](https://docs.djangoproject.com/en/2.2/topics/logging/#django-s-default-logging-configuration)).
+To make sure that errors get logged appropriately in Heroku, set the following
+baseline logging settings in your `settings.py` file:
+
+```python
+import os
+
+LOGGING = {
+    'version': 1,
+    'disable_existing_loggers': False,  # Preserve default loggers
+    'handlers': {
+        'console': {
+            'class': 'logging.StreamHandler',
+        },
+    },
+    'loggers': {
+        'django': {
+            'handlers': ['console'],
+            'level': os.getenv('DJANGO_LOG_LEVEL', 'INFO'),
+        },
+    },
+}
+```
+
+For more detail on Django's logging framework, [see the documentation](https://docs.djangoproject.com/en/2.2/topics/logging/).
+
+## Provision Heroku resources
+
+Once your application is properly configured for Heroku, the following instructions
+will help you deploy your application to the platform.
+
+### Ensure that the Heroku CLI is installed
+
+The fastest way to get a project up and running on Heroku is to use the
+[Heroku CLI](https://devcenter.heroku.com/articles/heroku-cli). Before you start,
+make sure you have the CLI installed locally. In addition, confirm that you have the
+manifest plugin) installed:
+
+```
+heroku plugins | grep manifest
+```
+
+If you don't see any output, [follow the official instructions for installing the
+plugin](https://devcenter.heroku.com/changelog-items/1441).
+
+### Create Heroku config files
+
+Define config files relative to the root of your repo.
+
+#### `heroku.yml`
+
+The `heroku.yml` manifest file tells Heroku how to build and network the services
+and containers that comprise your application. This file should live in the root of your project repo.
+For background and syntax, see [the
+documentation](https://devcenter.heroku.com/articles/build-docker-images-heroku-yml).
+Use the following baseline to get started:
+
+```yml
+# Define addons that you need for  your project, such as Postgres, Redis, or Solr.
+setup:
+  addons:
+    - plan: heroku-postgresql
```

Comment:
If your application requires PostGIS, [you need to manually install it in your database](https://devcenter.heroku.com/articles/postgis).

```bash
heroku psql -a <YOUR_APP>
create extension postgis;
```

The DATABASE_URL environmental variable will always use the Postgres engine (`postgres://...`), so you also need to override this in your Django settings file [like this](https://github.com/datamade/la-metro-councilmatic/pull/529/files/e4f8ecefdcce4007be4dc58fcd628d9436765126#diff-70d20b507e138d0a724e21e9043c17c0).

Reply:
Also, for advanced use cases, here are docs on:

- Sharing a database between applications (e.g., a scraping dashboard and a web application): https://devcenter.heroku.com/articles/heroku-postgresql#sharing-heroku-postgres-between-applications
- Interacting with your database remotely (e.g., populating your app from a remote scraping process): https://devcenter.heroku.com/articles/connecting-to-heroku-postgres-databases-from-outside-of-heroku

Reply:
Thanks! I added docs for PostGIS to the `Enable additional services` section, and added the two links above as sections in `Troubleshooting`.